multicluster/controllers/multicluster/commonarea/resourceimport_controller.go

```diff
@@ -239,15 +249,14 @@ func (r *ResourceImportReconciler) handleResImpDeleteForService(ctx context.Cont
 	err = r.localClusterClient.Get(ctx, svcName, svc)
 	if err != nil {
 		if apierrors.IsNotFound(err) {
+			klog.V(2).InfoS("Service corresponding to ResourceImport has already been deleted",
+				"service", svcName.String(), "resourceimport", klog.KObj(resImp))
 			return cleanupServiceImport()
 		}
 		return ctrl.Result{}, err
 	}
 	err = r.localClusterClient.Delete(ctx, svc, &client.DeleteOptions{})
 	if err != nil {
-		if apierrors.IsNotFound(err) {
```

Comment:
The reason I'm deleting this condition branch is because I believe that this has too little a chance of happening to be handled differently. If you look at the lines before, it is only when the service is deleted in the milliseconds between the service GET and DELETE calls that this branch will ever be true... Plus, it seems that when the err is not "notFound" we are skipping the ServiceImport cleanup